test/cql-pytest/test_permissions.py

```diff
@@ -553,3 +554,174 @@ def test_native_functions_always_exeutable(cql):
+# access to the table it just created (we call this feature "auto-grant").
+# The following tests check the auto-grant feature. The first test checks
+# that the base table created by CREATE TABLE is usable by the role creating
+# it. The second test is about the view created by CREATE MATERIALIZED VIEW
+# being accessible. The third test is about newly created CDC being accessible.
+def test_auto_grant_base(cql, test_keyspace):
+    with new_user(cql) as username:
+        # Grant username CREATE permissions on test_keyspace, but no other
+        # permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, username)
+        with new_session(cql, username) as user_session:
+            schema = "p int primary key"
+            with new_test_table(user_session, test_keyspace, schema) as table:
+                # Check that the username was auto-granted SELECT permissions
+                user_session.execute(f'SELECT * FROM {table}')
+                # Check that the username was auto-granted MODIFY permissions
+                user_session.execute(f'INSERT INTO {table}(p) VALUES(42)')
+                # Check that the username was auto-granted ALTER permissions
+                user_session.execute(f"ALTER TABLE {table} WITH comment = 'hey'")
+                # When this scope ends, the test table is dropped by
+                # user_session, so we check that DROP permissions were auto-
+                # granted as well.
+
+# Test that a role creating a materialized view is automatically granted
+# permissions to read it.
+def test_auto_grant_view(cql, test_keyspace):
+    with new_user(cql) as username:
+        # Grant username CREATE permissions on test_keyspace, but no other
+        # permissions:
+        grant(cql, 'CREATE', 'KEYSPACE ' + test_keyspace, username)
+        with new_session(cql, username) as user_session:
+            schema = "p int primary key, v int"
+            with new_test_table(user_session, test_keyspace, schema) as table:
+                # Because username has CREATE permissions, it can also use
+                # CREATE MATERIALIZED VIEW
+                mv = f'{test_keyspace}.{unique_name()}'
+                user_session.execute(f'CREATE MATERIALIZED VIEW {mv} AS SELECT * FROM {table} WHERE p IS NOT NULL AND v IS NOT NULL PRIMARY KEY (v, p)')
+                # Check that the username was auto-granted SELECT permissions
+                # on the newly created materialized view:
+                user_session.execute(f'SELECT * FROM {mv}')
+                # Check that the username was auto-granted ALTER permissions
+                # on the materialized view
+                user_session.execute(f"ALTER MATERIALIZED VIEW {mv} WITH comment = 'hey'")
+                # Check that the username was auto-granted DROP permissions
+                # on the materialized view
+                user_session.execute(f'DROP MATERIALIZED VIEW {mv}')
```

Comment:
That's a bit surprising, I'd expect MV to be also automatically dropped, just as is the case for the table in the testcase above.

Reply:
Actually, you need to to drop a table's materialized views before you're allowed to drop the base table - if you don't, you can't drop the base table. It's not automatic.
What I could have done, I guess, was to write a contextmanager to do this automatically.